src/services/transactions_queued.rs

```diff
@@ -0,0 +1,215 @@
+use crate::config::{base_transaction_service_url, request_cache_duration};
+use crate::models::backend::transactions::MultisigTransaction;
+use crate::models::commons::{Page, PageMetadata};
+use crate::models::service::transactions::summary::{ConflictType, Label, TransactionListItem};
+use crate::providers::info::{DefaultInfoProvider, InfoProvider};
+use crate::services::offset_page_meta;
+use crate::utils::cache::CacheExt;
+use crate::utils::context::Context;
+use crate::utils::errors::ApiResult;
+use itertools::Itertools;
+
+// use https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.peekable
+pub fn get_queued_transactions(
+    context: &Context,
+    safe_address: &String,
+    page_url: &Option<String>,
+    timezone_offset: &Option<String>,
+    trusted: &Option<bool>,
+) -> ApiResult<Page<TransactionListItem>> {
+    let mut info_provider = DefaultInfoProvider::new(context);
+
+    // Parse page meta (offset and limit)
+    let page_meta = PageMetadata::from_url_string(page_url.as_ref().unwrap_or(&"".to_string()));
+    // Adjust the page meta to fetch additional information of adjacent pages
+    let adjusted_page_meta = adjust_page_meta(&page_meta);
+
+    // Allow to also query queued transactions that are not submitted by an owner or delegate
+    let display_trusted_only = trusted.unwrap_or(true);
+
+    // As we require the Safe nonce later we use it here explicitely to query transaction that are in the future
+    let safe_nonce = info_provider.safe_info(safe_address)?.nonce as i64;
+    let url = format!(
+        "{}/v1/safes/{}/multisig-transactions/?{}&nonce__gte={}&ordering=+nonce&trusted={}",
+        base_transaction_service_url(),
+        safe_address,
+        adjusted_page_meta.to_url_string(),
+        safe_nonce,
+        display_trusted_only
+    );
+
+    let body = context
+        .cache()
+        .request_cached(&context.client(), &url, request_cache_duration())?;
+    let mut backend_transactions: Page<MultisigTransaction> = serde_json::from_str(&body)?;
+
+    // If there is a next url we remove the last item for information on the next page
+    let after_tx = if backend_transactions.next.is_some() {
```

Comment:
Here I would use `and_then` : https://doc.rust-lang.org/std/option/enum.Option.html#method.and_then

Reply:
yes would make this a little nicer